### PR TITLE
Bug: fix failed l1 message tx replay

### LINKED ISSUE #1504: bug: l1 message is processed repeatedly
If an L1 message fails to be processed, it gets processed again when you restart the node.

### PR DESCRIPTION
**Pull request by prtk418**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Bugfix

Resolves: [#1504](https://github.com/keep-starknet-strange/madara/issues/1504)

### CODE REVIEW

File: crates/pallets/starknet/src/lib.rs [hunk 1]
```diff
@@ -664,36 +664,41 @@ pub mod pallet {
...
+                    );
+                    Ok(())
+                }
+                Err(e) => {
+                    // commit nonce even if txn fails to avoid replays
+                    storage::transactional::with_transaction(|| {
+                        storage::TransactionOutcome::Commit(Result::<_, DispatchError>::Ok(
+                            L1Messages::<T>::mutate(|nonces| nonces.insert(nonce))
+                        ))
+                    })?;
```
Comment (prtk418) [inline crates/pallets/starknet/src/lib.rs:697]: @apoorvsadana Committing nonce data separately for error case, however it still doesn't commit to db. Can you please check once and help me out here.

**Comment by prtk418** [comment]
PR on hold because the issue might potentially get fixed by a rebase on progress by @tdelabro

**Comment by tdelabro** [comment]
@prtk418 #1566 was merged, can you rebase yourself and check if the issue is still there and if so, fix it

**Comment by tdelabro** [comment]
@prtk418 are you still working on this?

**Comment by prtk418** [comment]
> @prtk418 are you still working on this?

Hi @tdelabro , yes I will be picking this back up.

**Comment by tdelabro** [comment]
repository archived in favor of https://github.com/madara-alliance/madara

**Pull request closed without merging**